zookeeper-server/zookeeper-server-3.5/src/main/java/com/yahoo/vespa/zookeeper/VespaZooKeeperServerImpl.java

```diff
@@ -76,6 +79,41 @@ private String transformConfigToString(ZookeeperServerConfig config) {
         sb.append("serverCnxnFactory=org.apache.zookeeper.server.NettyServerCnxnFactory").append("\n");
         ensureThisServerIsRepresented(config.myid(), config.server());
         config.server().forEach(server -> addServerToCfg(sb, server));
+        sb.append(createTlsQuorumConfig(config));
+        return sb.toString();
+    }
+
+    private String createTlsQuorumConfig(ZookeeperServerConfig config) {
+        StringBuilder sb = new StringBuilder();
+
+        // Common config
+        sb.append("ssl.quorum.hostnameVerification=false\n");
+        sb.append("ssl.quorum.clientAuth=NEED\n");
+        sb.append("ssl.quorum.ciphersuites=").append(String.join(",", new TreeSet<>(TlsContext.ALLOWED_CIPHER_SUITES))).append("\n");
+        sb.append("ssl.quorum.enabledProtocols=").append(String.join(",", new TreeSet<>(TlsContext.ALLOWED_PROTOCOLS))).append("\n");
+        sb.append("ssl.quorum.protocol=TLSv1.2\n");
```

Comment:
This should set to 'TLS'. It's effectively setting max TLS version, disabling TLSv1.3 even when it's listed in `enableProtocols`. See table in https://bugs.openjdk.java.net/browse/JDK-8202625.

Reply:
ZooKeeper doc just says that 'TLSv1.2' is default value, but not anything about other possible values. But this makes sense, thanks, will fix